tests/test_single_sbatch_runner.py

```
@@ -487,6 +487,20 @@ def test_dse_and_non_dse(self, nccl_tr: TestRun, slurm_system: SlurmSystem) -> N
         assert runner.on_job_submit.call_count == 3  # 2 dse runs + 1 non-dse run
 
 
+def test_rejects_single_sbatch_incompatible(nccl_tr: TestRun, slurm_system: SlurmSystem) -> None:
+    cast(NCCLTestDefinition, nccl_tr.test).cmd_args.use_deepep_matrix = True
+    tc = TestScenario(name="tc", test_runs=[nccl_tr])
+    runner = SingleSbatchRunner(mode="run", system=slurm_system, test_scenario=tc, output_path=slurm_system.output_path)
+
+    with pytest.raises(ValueError) as exc_info:
```

Ruff (0.15.20)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)